frontend/src/pages/trucks/components/add-truck-form/add-truck-form.tsx

```diff
@@ -0,0 +1,39 @@
+import { Form } from '~/libs/components/components.js';
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import { useAppDispatch, useCallback } from '~/libs/hooks/hooks.js';
+import { type TruckFormModel } from '~/packages/trucks/libs/types/types.js';
+import { truck } from '~/packages/trucks/libs/validation-schemas/validation-schemas.js';
+import { actions as truckActions } from '~/slices/trucks/trucks.js';
+
+import { DEFAULT_TRUCK_PAYLOAD } from './libs/constants/constants.js';
+import { ADD_TRUCK_FIELDS } from './libs/fields/add-truck.fields.js';
+import { getTruckEntity } from './libs/helpers/get-truck-entity.helper.js';
+import styles from './styles.module.scss';
+
+const AddTruckForm: React.FC = () => {
+  const dispatch = useAppDispatch();
+
+  const handleFormSubmit = useCallback(
+    (payload: TruckFormModel): void => {
+      void dispatch(truckActions.addTruck(getTruckEntity(payload)));
+    },
+    [dispatch],
+  );
+
+  return (
+    <div className={styles.formWrapper}>
+      <h3 className={getValidClassNames('h4', 'uppercase', styles.title)}>
+        Add Truck
+      </h3>
+      <Form
+        fields={ADD_TRUCK_FIELDS}
+        defaultValues={DEFAULT_TRUCK_PAYLOAD}
+        validationSchema={truck}
```

Comment:
actually don't understand why do you pass here **truck** validation schema that you've created on fe, can't you reuse **truckCreateRequestBody** from the shared?